framework/web/Request.php

```diff
@@ -196,6 +243,34 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the trusted host config.
+     * @param array $headers
+     * @since 2.0.13
+     * @return array filtered headers.
+     */
+    protected function filterHeaders(array $headers)
+    {
+        $host = $this->getUserHost();
+        $ip = $this->getRemoteIP();
+        foreach ($this->trustedHostConfig as $hostRegex => $trustedHeaders) {
+            if (!is_array($trustedHeaders)) {
+                $hostRegex = $trustedHeaders;
+                $trustedHeaders = $this->secureHeaders;
+            }
+
+            if (preg_match($hostRegex, $host) || preg_match($hostRegex, $ip)) {
+                foreach ($this->secureHeaders as $secureHeader) {
+                    if (!in_array($secureHeader, $trustedHeaders)) {
+                        unset($headers[$secureHeader]);
+                    }
+                }
+                return $headers;
+            }
+        }
+        return [];
```

Comment:
this would mean all headers are filtered that are not in trusted headers? That does not make sense. Headers should always be available, but trusted headers should be filtered if sent by ips that are not trusted.